src/praisonai-agents/praisonaiagents/skills/prompt.py

```
@@ -37,29 +38,37 @@ def format_skill_for_prompt(skill: SkillMetadata) -> str:
   </skill>"""
 
 
-def generate_skills_xml(skills: List[SkillMetadata], working_directory: str = None) -> str:
+def generate_skills_xml(skills: List[SkillMetadata], working_directory: str = None, budget: Optional[SkillPromptBudget] = None) -> str:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)